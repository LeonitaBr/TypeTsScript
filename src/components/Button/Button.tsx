import React, {memo} from 'react';
import {Text, TouchableOpacity} from 'react-native';
import styles from './style';

interface ButtonProps {
  onPressButton: Function;
  title: string;
}

const Button: React.FC<ButtonProps> = ({onPressButton, title}) => {
  return (
    <TouchableOpacity
      onPress={() => onPressButton()}
      style={styles.btnContainer}>
      <Text style={styles.btnText}>{title}</Text>
    </TouchableOpacity>
  );
};
export default memo(Button);
