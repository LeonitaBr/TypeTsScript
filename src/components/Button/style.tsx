import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  btnContainer: {
    // width: 100,
    height: 50,
    // marginHorizontal: 10,
    backgroundColor: '#DB7093',
    justifyContent: 'center',
    alignItems: 'center',
  },
  btnText: {
    colot: 'white',
    fontSize: 16,
  },
});
export default styles;
