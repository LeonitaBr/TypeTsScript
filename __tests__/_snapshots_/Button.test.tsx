import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../src/components/Button/Button';

describe('Some component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button onPressButton={() => {}} title={'AAAA'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
