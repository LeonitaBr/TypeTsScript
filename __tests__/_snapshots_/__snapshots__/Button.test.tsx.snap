// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Some component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#DB7093",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "colot": "white",
        "fontSize": 16,
      }
    }
  >
    AAAA
  </Text>
</View>
`;
